cartridges/int_algolia_sfra/cartridge/static/default/js/utils/productTileUtils.js

```diff
@@ -0,0 +1,43 @@
+var lastRequestTime = '';
+
+
+// Listen for AJAX requests
+// If the request is an add to cart request, check if the last request was less than 4 seconds ago
+// If it was, abort the request 
+// Workaround to not modifying orginal cartridge code to fix issue with multiple add to cart requests because of adding productTile to PDP page
+$(document).ajaxSend(function(event, jqxhr, settings) {
+
+    if (settings.url.indexOf('Cart-AddProduct') === -1) {
+        return;
+    }
+
+    var now = Date.now();
+
+    if (lastRequestTime && now - lastRequestTime < 4) {
```

Comment:
`Date.now()` returns milliseconds, so the check will become true after 4 ms only.

Reply:
I intend to prevent double events by implementing a check like that.

Reply:
I know, but your code doesn't match the comment above. Comment says that you want to check _"if the last request was less than 4 seconds ago"_, but in the code you only wait 4ms

Reply:
ahh, my fault, I am fixing the comment now